### PR TITLE
Batch and error check allocations

### PR DESCRIPTION
**Pull request by Starbuck5**
I was looking into optimization opportunities and found some allocation requests (mallocs and PyMem_News) right next to each other, and I thought surely calling the function once will be faster than calling it twice. It is also simpler to check if one buffer failed to allocate, and simpler to free one buffer at the end, rather than 2 or 3.

Looking through the code I also saw that the malloc statements in the blur routines weren't error checked at all, so I fixed that.

Running very favorable microbenchmarks I saw very small performance increases, like 1-3%, or no impact. I'm unsure if running a tight loop over identical calls reduces the execution time burden of malloc / PyMem_New, so maybe performance would be affected differently in a more realistic scenario?

Anyways since this is a small thing I'm not trying to pitch this as a performance PR, instead this is a code quality PR. We want our code to be the best it can be, and I think that reducing the number of allocation calls helps that goal.

### CODE REVIEW

File: src_c/transform.c [hunk 1]
```diff
@@ -3574,9 +3574,17 @@ box_blur(SDL_Surface *src, SDL_Surface *dst, int radius, SDL_bool repeat)
...
-    Uint32 *sum_h = malloc(nb * sizeof(Uint32));
+
+    // Allocate bytes for buf, sum_v, and sum_h at once to reduce allocations.
+    Uint32 *overall_buf = malloc(sizeof(Uint32) * (dst_pitch * 2 + nb));
+    Uint32 *buf = overall_buf;
+    Uint32 *sum_v = overall_buf + dst_pitch;
+    Uint32 *sum_h = overall_buf + dst_pitch * 2;
+
+    if (overall_buf == NULL) {
+        PyErr_NoMemory();
```
Comment (ankith26) [inline src_c/transform.c:3585]: This function is called within `Py_BEGIN_ALLOW_THREADS`/`Py_END_ALLOW_THREADS` so I don't think we are allowed to call this here as it is python API.

Comment (Starbuck5) [reply src_c/transform.c]: Good catch! Fixed.

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :tada:

**Review by bilhox — CHANGES_REQUESTED** [review]
LGTM, just a little thing to fix, but otherwise it seems fair.

**Review by bilhox — APPROVED** [review]
LGTM

**Pull request merged**